clicky-core/src/devices/display/hd66753.rs

```diff
@@ -155,15 +157,25 @@ impl Hd66753 {
             };
 
             let cgram_window = cgram
-                .chunks_exact(EMU_CGRAM_WIDTH * 2 / 8 / 2)
-                .take(height)
-                .flat_map(|row| row.iter().take(CGRAM_WIDTH * 2 / 8 / 2).rev());
+                    .chunks_exact(EMU_CGRAM_WIDTH * 2 / 8 / 2)
+                    .take(height)
+                    .flat_map(|row| {
+                        match ireg.sgs {
+                            true => Either::Left(row.iter().take(CGRAM_WIDTH * 2 / 8 / 2)),
+                            false => Either::Right(row.iter().take(CGRAM_WIDTH * 2 / 8 / 2).rev()),
+                        }
+                        
+                     });
 
             // TODO: implement cursor control
 
             let new_buf = cgram_window.flat_map(|w| {
                 // every 16 bits = 8 pixels
-                (0..8).rev().map(move |i| {
+                let bit_iter = match ireg.sgs {
+                    true => Either::Left((0..8)),
+                    false => Either::Right((0..8).rev()),
+                };
+                bit_iter.map(move |i| {
```

Comment:
```suggestion
                (0..8).map(move |i| {
                    let i = if ireg.sgs { i } else { 7 - i };
```

I think this lets you avoid this Either?

Reply:
Yup, looks better. Applied.